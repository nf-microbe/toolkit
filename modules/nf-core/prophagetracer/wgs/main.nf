process PROPHAGETRACER_WGS {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/blast_bwa_sambamba_samtools_gawk:6d2fecd74dcf1c21' :
        'community.wave.seqera.io/library/blast_bwa_sambamba_samtools_gawk:edd6b16d4dd01302' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(bam)

    output:
    tuple val(meta), path("${prefix}.prophage.out") , emit: prophage
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    def is_compressed = fasta.getExtension() == "gz" ? true : false
    def fasta_name = is_compressed ? fasta.getBaseName() : fasta
    """
    if [ "${is_compressed}" == "true" ]; then
        gzip -c -d ${fasta} > ${fasta_name}
    fi

    samtools view ${bam} -o ${prefix}.sam

    prophage_tracer_WGS.sh \\
        -m ${prefix}.sam \\
        -r ${fasta_name} \\
        -p ${prefix} \\
        -t ${task.cpus} \\
        ${args} || if grep -q "SR_evidence.list1" .command.log; then true; else false; fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        prophagetracer: \$(echo \$( prophage_tracer_WGS.sh -h | sed -n "s/Prophage Tracer V//; s/\s.*//;  2p" ))
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.prophage.out

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        prophagetracer: \$(echo \$( prophage_tracer_WGS.sh -h | sed -n "s/Prophage Tracer V//; s/\s.*//;  2p" ))
    END_VERSIONS
    """
}
