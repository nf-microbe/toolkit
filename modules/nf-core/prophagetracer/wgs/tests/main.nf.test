nextflow_process {

    name "Test Process PROPHAGETRACER_WGS"
    script "../main.nf"
    process "PROPHAGETRACER_WGS"

    tag "modules"
    tag "modules_nfcore"
    tag "prophagetracer"
    tag "prophagetracer/wgs"

    test("fasta.gz - bam") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/np_salmoLT2.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/bam/test1.markdup.bam", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - bam - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/np_salmoLT2.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/bam/test1.markdup.bam", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
