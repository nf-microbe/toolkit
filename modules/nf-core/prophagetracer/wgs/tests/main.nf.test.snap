{
    "fasta.gz - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ],
                "prophage": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-16T11:27:20.147806848"
    },
    "fasta.gz - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,8166ad81eb844e4172fd02fb2ec0db2f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ],
                "prophage": [
                    [
                        {
                            "id": "test"
                        },
                        "test.prophage.out:md5,8166ad81eb844e4172fd02fb2ec0db2f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e4ec57dd9ee845f5409bc7f06d89bfaf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T09:42:12.105261789"
    }
}